docs/developer-docs/gas-cost.md

```diff
@@ -1,100 +1,230 @@
+- If subnet usage is above `450GiB`, then the amount of reserved cycles per allocated byte grows linearly depending on the subnet usage, from `0` to `10` years worth of storage payments at the subnet capacity (which is currently `750GiB`).
 
 A controller of a canister can disable resource reservation by setting the `reserved_cycles_limit=0` in canister settings.
-Such opted-out canisters would not be able to allocate if the subnet usage is above `450GiB` though.
\ No newline at end of file
+Such opted-out canisters would not be able to allocate if the subnet usage is above `450GiB` though.
+
+### Special features
+
+Special features aside from messaging, storage, and execution resources have different costs since they use special infrastructure and architectures to provide the feature's functionality. These special features include:
+
+- **HTTPS outcalls**: The cost of HTTPS outcalls is calculated using a unique formula since the feature includes a quadratic component in its implementation. The cost for an HTTPS outcall is calculated using the formula `(3_000_000 + 60_000 * n) * n` for the base fee and `400 * n` each request byte and `800 * n` for each response byte, where `n` is the number of nodes in the subnet. These costs are included in the chart found below.
+
+- **Bitcoin API**: Pricing for the **Bitcoin API** is available in the [Bitcoin API documentation](./integrations/bitcoin/bitcoin-how-it-works.md).
+
+- **Chain-key signing API**: Pricing for the **Chain-key signing API** is available in the [Chain-key signing / threshold ECDSA documentation](./integrations/t-ecdsa/t-ecdsa-how-it-works.md).
+
+### Replication
+
+Replication refers to the number of times a canister is replicated, which will depend on the subnet that the canister is deployed to.
+
+#### Local development
+
+When a canister is deployed to a local development environment, the canister is deployed to a single node (the locally running instance of the replica). Cycles charged to locally deployed canisters have a cost that is 1/13th the cost when deployed to a 13-node subnet. In local development environments, cycles can be fabricated using dfx.
+
+#### Mainnet development
+
+On a 13-node subnet, the canister is running on 13 nodes and therefore is replicated 13 times, whereas on a 34-node subnet, the canister is running on 34 nodes and therefore is replicated 34 times. The cost of resources on a 13-node subnet is different than the cost of resources on a 34-node subnet.
+
+If you intend to deploy canisters on high-replication subnets, your canister should be prepared for an increase in cycles prices with an increase in the subnet's replication factor when the subnet grows over time. For this reason, it is recommended to attach more cycles to a call than the current price for a high-replication subnet of a given size suggests.
+
+## Who is responsible for paying cycles?
```

Comment:
Let's unify this with the messaging costs?

Reply:
I think it is important to keep this section independent in order to draw attention to the fact that there are instances where different canisters are responsible based on the operation.